drivers/dai/intel/hda/hda.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/logging/log.h>
+
+#define DT_DRV_COMPAT intel_hda_dai
+#define LOG_DOMAIN dai_intel_hda
+
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#include "hda.h"
```

Comment:
Why are you placing headers in sources directory and including by relative path?
Headers should be placed in the include\zephyr\drivers directory
and included by
#include <zephyr/drivers/intel/dai/hda.h>

Reply:
this is implementation specific header, drivers are using public dai.h header for access